cmd/argocd/commands/admin/settings_rbac.go

```diff
@@ -430,6 +430,9 @@ func checkPolicy(subject, action, resource, subResource, builtinPolicy, userPoli
 			log.Fatalf("could not set user policy: %v", err)
 			return false
 		}
+		if err := enf.CheckUserDefinedRoleReferentialIntegrity(); err != nil {
+			log.Fatal(err.Error())
+		}
```

Comment:
I think this should be a part of ValidatePolicy instead

Reply:
Updated per your suggestion